### PR TITLE
UX Improvements in `EditableTextField` and `EditableTextArea` fields

### PR DESCRIPTION
**Pull request by amar-hunar**
This PR introduces UX improvements in `EditableTextArea` and `EditableTextField`

## Features:

- Update the edit UX for `EditableTextArea` and `EditableTextField`. If the user clicks out (onBlur) of the input, the unsaved edit changes should be discarded

## Links:
- [Storybook Link](https://ux-improvements--666ac08efe69dd9c59a1e4c6.chromatic.com/?path=/story/components-editabletextarea--playground)

### CODE REVIEW

File: src/components/EditableTextArea/EditableTextArea.tsx [hunk 1]
```diff
@@ -68,9 +72,12 @@ export const EditableTextArea = ({
...
         }
 
-        if (value && hasErrors) {
-            setEditedValue(value);
-            setIsEditing(false);
+        if (value) {
+            const timer = setTimeout(() => {
+                setEditedValue(value);
+                setIsEditing(false);
+            });
```
Comment (misss-popcorn) [inline src/components/EditableTextArea/EditableTextArea.tsx:79]: where is the time?

File: src/components/EditableTextField/EditableTextField.tsx [hunk 2]
```diff
@@ -80,9 +84,12 @@ export const EditableTextField = ({
...
         }
 
-        if (value && hasErrors) {
-            setEditedValue(value);
-            setIsEditing(false);
+        if (value) {
+            const timer = setTimeout(() => {
+                setEditedValue(value);
+                setIsEditing(false);
+            });
```
Comment (misss-popcorn) [inline src/components/EditableTextField/EditableTextField.tsx:91]: why are we setting timeout without time?

File: src/components/EditableTextArea/EditableTextArea.tsx (hunk 1 above)
Comment (amar-hunar) [reply src/components/EditableTextArea/EditableTextArea.tsx:79]: It is intentionally left out. If we do not pass a time, the callback is called in the next event cycle and that is the intended behaviour here 

Reference:
- [MDN](https://developer.mozilla.org/en-US/docs/Web/API/setTimeout#delay)

File: src/components/EditableTextField/EditableTextField.tsx (hunk 2 above)
Comment (amar-hunar) [reply src/components/EditableTextField/EditableTextField.tsx:91]: It is intentionally left out. If we do not pass a time, the callback is called in the next event cycle and that is the intended behaviour here 

Reference:
- [MDN](https://developer.mozilla.org/en-US/docs/Web/API/setTimeout#delay)

File: src/components/EditableTextArea/EditableTextArea.tsx (hunk 1 above)
Comment (amar-hunar) [reply src/components/EditableTextArea/EditableTextArea.tsx:79]: Removed this login in next PR #15

File: src/components/EditableTextField/EditableTextField.tsx (hunk 2 above)
Comment (amar-hunar) [reply src/components/EditableTextField/EditableTextField.tsx:91]: Removed this logic in next PR #15

**Pull request merged**